api/v1alpha1/certificatestore_types.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CertificateStoreSpec defines the desired state of CertificateStore
+type CertificateStoreSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Name of the certificate store provider
+	Provider string `json:"provider,omitempty"`
+
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// Parameters of the certificate store
+	Parameters runtime.RawExtension `json:"parameters,omitempty"`
+}
+
+// CertificateStoreStatus defines the observed state of CertificateStore
+type CertificateStoreStatus struct {
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
```

Comment:
What are these comments for?

Reply:
These annotations are here so kubebuilder could generate the yaml files based on the spec defined in go files.